vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/view/launch/InputLayout.java

```diff
@@ -1,334 +1,525 @@
+                            .append(disablingInputNames)
+                            .append(".</li>");
                 }
-                else {
-                    listLayout.addMember(new ListHLayout(listLayout, true, "", true));
-                    startItem.clearValue();
-                    stopItem.clearValue();
-                    stepItem.clearValue();
+                if(requiredInputNames.size() > 0){
+                    disabledHoverMessage.append("<li>following input(s) are empty: ")
+                            .append(requiredInputNames)
+                            .append(".</li>");
                 }
-            }  
-        });
-     
-        cbOptionalInputForm = FieldUtil.getForm(cbOptionalInputItem);        
-        cbOptionalInputForm.setTop(0);
-        cbOptionalInputForm.setHeight(15);
-        cbOptionalInputForm.setWidth(25);
-        cbOptionalInputForm.setMaxWidth(25);
-        cbOptionalInputForm.setMaxHeight(25);
-        cbOptionalInputForm.setPadding(0);
-        cbOptionalInputForm.setCellPadding(0);
-        cbOptionalInputForm.setMargin(0);
-        sourceFieldHLayout.addMember(cbOptionalInputForm);
-    }
-
-    private void setList() {
-
-        fieldHLayout.addMember(listLayout);
-        fieldHLayout.removeMember(startItemForm);
-        fieldHLayout.removeMember(stopItemForm);
-        fieldHLayout.removeMember(stepItemForm);
-    }
-
-    private void setRange() {
-
-        fieldHLayout.removeMember(listLayout);
-        fieldHLayout.addMember(startItemForm);
-        fieldHLayout.addMember(stopItemForm);
-        fieldHLayout.addMember(stepItemForm);
-    }
-
-    @Override
-    public boolean validate() {
-
-        InputType type = InputType.valueOf(selectItem.getValueAsString());
-        if (type == InputType.List) {
-            boolean valid = true;
-            for (Canvas canvas : listLayout.getMembers()) {
-                if (canvas instanceof ListHLayout) {
-                    ListHLayout item = (ListHLayout) canvas;
-                    if (!item.validate()) {
-                        valid = false;
-                    }
+                if(disablingValues.size() > 0) {
+                    disablingValues.forEach(
+                            (inputName, value) -> disabledHoverMessage.append("<li>input [")
+                                    .append(inputName)
+                                    .append("] has value ")
+                                    .append(value)
+                                    .append(".</li>"));
+                }
+                for (FormItem field : this.masterForm.getFields()) {
+                    field.disable();
+                    field.setPrompt(disabledHoverMessage.toString());
                 }
             }
-            return valid;
-        } 
-        else {
-                    return startItem.validate() & stopItem.validate() & stepItem.validate();
+            if (this.isMasterEmpty() != wasEmpty){
```

Comment:
No, if this methods enabled or disabled this input and its value is/was not empty.
For instance, lets say B (this) is disabled by A, which is empty. When user fills A, A fires B.checkDependencies, which allows B to see that is must disable its field. Once that's done, B's value went from whatever it was to nothing (disabled), so we need to apply B.onValueChanged. Because if B was disabling C, now that B is disabled we must re-enable C.

Reply:
Ok I get it. I did not understand because there is no code that changes value in this PR, so I thought that the change could only come from the user. But indeed, the enabling/disabling of input can change the "virtual" value through the `formValue()` method. 
So it's good.